src/api/workflow.py

```diff
@@ -0,0 +1,570 @@
+        """Validate a workflow definition.
+
+        Args:
+            workflow (Union[str, dict]): If str, this is assumed to be the path to the workflow definition json file; If dict, this is assumed to be loaded from the workflow json definition.
+            verbose (bool, optional): Verbose output for validate. Defaults to False.
+
+        Returns:
+            dict: `dict` with the following keys:
+                - `workflow_validity`: bool flag of the validity of the workflow definition
+                - `detail`: detailed info about the validity check.
+        """
+
+        temp_workflow_engine = Workflow.create_workflow_engine(workflow)
+        return temp_workflow_engine.validate()
+
+    def validate(self, verbose=False) -> dict:
+        """Validate `self.workflow`
+
+        Args:
+            verbose (bool, optional): Verbose output for validate. Defaults to False.
+
+        Returns:
+            dict: `dict` with the following keys:
+                - `workflow_validity`: bool flag of the validity of the workflow definition
+                - `detail`: detailed info about the validity check.
+        """
+        return self.workflow_engine.validate()
+
+    def save(self, json_path: str = None) -> None:
+        """Save the workflow as a json file.
+
+        Args:
+            json_path (str, optional): path to the file to be saved. Defaults to None.
+        """
+
+        if self.workflow_engine is None:
+            logging.warning("self.workflow_engine is None, there is nothing to save.")
+        else:
+            if isinstance(json_path, str):
+                self.workflow_engine.save_workflow(json_path)
+            else:
+                logging.error("json_path needs to be str")
+
+    def run_workflow(self, verbose: bool = False) -> bool:
+        """Execute the workflow defined in self.workflow
+
+        Args:
+            verbose (bool, optional): bool. Wether to output detailed information. Defaults to False.
+
+        Returns:
+            bool: Whether the run is successful or not.
+
+        """
+        self.workflow_engine.run_workflow(verbose)
+
+
+class MethodCall:
+    """The MethodCall State class. This class also covers the `Embedded MethodCall` state type.
+    A typical use case of execute a MethodCall state would be: `self.payloads = MethodCall(state_dict, self.payloads).run().get_payloads()`
+    """
+
+    def __init__(self, state_dict, payloads):
+        self.payloads = payloads  # self.payloads takes in the input payloads, add key-value pairs contents based on state_dict['Payloads'], and will be available by self.get_payloads()
+        self.parameters = {}
+        self.state_dict = state_dict
+        self.dollar = None  # the return of the current method call will be stored in this variable.
+        self.build_parameters(state_dict)
+
+    def build_parameters(self, state_dict):
+        if isinstance(state_dict["Parameters"], dict):
```

Comment:
Missing doc strings, and args type docs.

Reply:
Same for the functions below.

Reply:
Other than the WorkflowAPI class, other classes are internal facing for now